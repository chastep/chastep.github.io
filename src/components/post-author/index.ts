export { PostAuthor } from "./post-author";
