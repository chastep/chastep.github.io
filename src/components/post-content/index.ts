export { PostContent } from "./post-content";
