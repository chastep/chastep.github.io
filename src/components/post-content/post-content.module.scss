@use "src/assets/styles/mixins";
@use "src/assets/styles/variables";

.postContent {
  max-width: variables.$layout-post-single-width;
  @include mixins.margin-auto();
  @include mixins.padding(0, 0.5);

  .title {
    font-size: variables.$typography-base-font-size * 2;
    font-weight: 600;
    margin-left: auto;
    margin-right: auto;
    max-width: variables.$layout-post-width;
    text-align: center;
    @include mixins.line-height(1.65);
    @include mixins.margin-top(1);
    @include mixins.margin-bottom(0);
  }

  .body {
    figure {
      @include mixins.margin-bottom(1);

      blockquote {
        font-style: italic;
        text-align: center;
        @include mixins.margin-top(0);
        @include mixins.padding(1, 0);

        p {
          font-size: variables.$typography-base-font-size * 1.6817;
          max-width: variables.$layout-post-width;
          @include mixins.margin-top(0);
          @include mixins.margin-bottom(1);
          @include mixins.line-height(1.5);
        }
      }
    }

    a {
      text-decoration: underline;
    }

    * {
      margin-left: auto;
      margin-right: auto;
      max-width: variables.$layout-post-width;
    }

    h2 > a {
      visibility: hidden;

      & > svg {
        fill: hsl(var(--color-foreground));
      }
    }

    img {
      max-width: 100%;
    }
  }
}

@include mixins.breakpoint-lg {
  .postContent {
    @include mixins.padding-equal(0);

    .title {
      font-size: variables.$typography-base-font-size * 3;
      @include mixins.line-height(2.25);
      @include mixins.margin-top(2.25);
      @include mixins.margin-bottom(1.5);
    }

    .body {
      font-size: variables.$typography-base-font-size * 1.125;
      @include mixins.line-height(1.125);
      @include mixins.margin-bottom(1.125);

      p {
        font-size: variables.$typography-base-font-size * 1.125;
        @include mixins.line-height(1.125);
        @include mixins.margin-bottom(1.125);
      }

      h2 > a {
        visibility: unset;
        @include mixins.padding-right(0.5);
      }
    }
  }
}
