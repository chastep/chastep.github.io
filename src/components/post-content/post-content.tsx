import React, { type FC } from "react";

import * as styles from "./post-content.module.scss";

interface PostContentProps {
  title: string;
  body: string;
}

const PostContent: FC<PostContentProps> = ({ body, title }) => (
  <div className={styles.postContent}>
    <h1 className={styles.title}>{title}</h1>
    <div className={styles.body} dangerouslySetInnerHTML={{ __html: body }} />
  </div>
);

export { PostContent };
