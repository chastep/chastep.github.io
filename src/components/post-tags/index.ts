export { PostTags } from "./post-tags";