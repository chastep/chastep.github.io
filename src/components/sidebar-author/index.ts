export { SidebarAuthor } from "./sidebar-author";