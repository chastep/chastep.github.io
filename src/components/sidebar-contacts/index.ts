export { SidebarContacts } from "./sidebar-contacts";
