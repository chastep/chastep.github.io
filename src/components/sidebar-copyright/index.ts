export { SidebarCopyright } from "./sidebar-copyright";
