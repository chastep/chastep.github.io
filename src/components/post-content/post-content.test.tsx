import React from "react";
import { test, describe, expect } from "bun:test";

import * as mocks from "@/mocks";
import { PostContent } from "@/components/post-content";
import { createSnapshotsRenderer } from "@/utils/render-with-coil-provider";

describe("PostContent", () => {
  test("renders correctly", () => {
    const props = {
      title: mocks.markdownRemark.frontmatter.title,
      body: mocks.markdownRemark.html,
    };

    const tree = createSnapshotsRenderer(<PostContent {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
