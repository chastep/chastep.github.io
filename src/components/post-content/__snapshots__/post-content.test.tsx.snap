// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`PostContent renders correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        "Perfecting the Art of Perfection",
      ],
      "props": {
        "className": undefined,
      },
      "type": "h1",
    },
    {
      "$$typeof": Symbol(react.test.json),
      "children": null,
      "props": {
        "className": undefined,
        "dangerouslySetInnerHTML": {
          "__html": "<p>An Essay on Typography by Eric Gill takes the reader back to the year 1930. The year when a conflict between two worlds came to its term. The machines of the industrial world finally took over the handicrafts.</p>",
        },
      },
      "type": "div",
    },
  ],
  "props": {
    "className": undefined,
  },
  "type": "div",
}
`;
